src/layering.jl

```diff
@@ -22,27 +22,36 @@ end
 function add_dummy_nodes!(graph, layer2nodes)
     dag_or_error(graph)
     nondummy_nodes = vertices(graph)
+    # mapping from edges in the original graph to paths in the graph with dummy nodes
+    edge_to_paths = Dict(e => eltype(graph)[] for e in edges(graph)) 
     node2layer = node2layer_lookup(layer2nodes)  # doesn't have dummy nodes, doesn't need them
     for cur_node in vertices(graph)
         cur_layer = node2layer[cur_node]
-        for out_node in copy(outneighbors(graph, cur_node))  # need to copy as outwise will mutate when the graph is mutated
+        for out_edge in filter(e -> e.src == cur_node, collect(edges(graph)))  # need to copy as outwise will mutate when the graph is mutated
```

Comment:
I had to change from iterating the outneighbors to iterating edges to make it work with WeightedGraphs, I couldn't find how to get back to the WeightedEdge with just the `src` and `dst` vertices in a generic way. Maybe there's a better way to do it ?

Reply:
i am not sure. I don't know light graphs that well

Aside: Apparently the rtight way to get the soource is not `e.src` but rather `src(e)`

Reply:
We don't need the `collect` if we are using `filter` as `filter` creates a `copy`

Reply:
> Apparently the rtight way to get the soource is not e.src but rather src(e)

fixed

> We don't need the collect if we are using filter as filter creates a copy

`filter` on a SimpleEdgeIter throws an MethodError, so I left the `collect` for now